lib/tests/csrfProtected.test.js

```diff
@@ -0,0 +1,152 @@
+import mockedAxios from "axios";
+import { createMocks } from "node-mocks-http";
+import { submitToAPI } from "../integration/helpers";
+import { logMessage } from "@lib/logger";
+import TestForm from "./testData";
+import { getCsrfToken } from "next-auth/client";
+import { csrfProtected } from "@lib/middleware/csrfProtected";
+
+jest.mock("axios");
+jest.mock("next-auth/client");
+
+const mockResponses = {
+  1: "test response", // TextField
+  2: "test response 2", // TextArea
+  4: "Manitoba", // DropDown
+  5: "Student", // Radio
+  6: [], // CheckBox
+  7: [
+    {
+      0: "Doe",
+      1: "John",
+      2: "CDS",
+      3: ["No"],
+    },
+    {
+      0: "Seat",
+      1: "Love",
+      2: "CHMC",
+      3: ["Yes", "Other"],
+    },
+  ],
+  8: "",
+};
+
+describe("Csrf Protection middleware", () => {
+  let mockedConsoleError;
+  beforeEach(() => {
+    mockedConsoleError = jest.spyOn(logMessage, "error");
+    mockedConsoleError.mockImplementation(() => {});
+  });
+  afterEach(() => {
+    mockedConsoleError.mockRestore();
+    getCsrfToken.mockRestore();
+  });
+
+  test("Should fail with code 405 Method Not Allowed", async () => {
```

Comment:
For example if the mock request is "GET" and only the "POST" path is CSRF protected then the status code should be 200.